lib/happo/public/happo-runner.js

```diff
@@ -162,6 +162,14 @@ window.happo = {
     box.right = Math.max(box.right, rect.right);
     box.top = Math.min(box.top, rect.top);
 
+    if (box.left < 0) {
+      box.left = 0; // Don't go outside window
+    }
+
+    if (box.top < 0) {
+      box.top = 0; // Don't go outside window
+    }
```

Comment:
I figured it was better in the js file because then we don't have to recompute `box.width` etc in ruby. I actually started with a solution in rb but ended up here. I think you're right about not doing it for every node though. I'll move it outside of the recursion.